src/renderer/wg_engine/tvgWgRenderer.cpp

```diff
@@ -175,7 +179,21 @@ RenderData WgRenderer::prepare(const Array<RenderData>& scene, RenderData data,
 }
 
 RenderData WgRenderer::prepare(Surface* surface, const RenderMesh* mesh, RenderData data, const RenderTransform* transform, Array<RenderData>& clips, uint8_t opacity, RenderUpdateFlag flags) {
-    return nullptr;
+    // get or create render data shape
+    auto renderDataShape = (WgRenderDataShape*)data;
+    if (!renderDataShape) {
+        renderDataShape = new WgRenderDataShape();
+        renderDataShape->initialize(mDevice);
+        renderDataShape->mPipelineBindGroupImage.initialize(mDevice, mPipelineImage, surface);
+    }
+    if (flags & (RenderUpdateFlag::Image | RenderUpdateFlag::Transform)) {
```

Comment:
++ RenderUpdateFlag::Color  //for opacity change.

Reply:
Good point. Fixed